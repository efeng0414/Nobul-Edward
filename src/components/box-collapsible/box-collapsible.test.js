import React from "react";
import renderer from "react-test-renderer";
import BoxCollapsible from "./index";

describe("BoxCollapsible", () => {
  const componentProps = {
    children: "abc",
    className: "component",
    isOpenByDefault: true
  };
  const componentJSX = <BoxCollapsible {...componentProps} />;
  describe("ComponentSnapshot", () => {
    const componentSnapshot = renderer.create(componentJSX).toJSON();
    it("should match existing snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });
  describe("Component Render", () => {
    it("should check if default props are set", () => {
      const defaultProps = {
        isOpenByDefault: false
      };
      const props = Enzyme.mount(<BoxCollapsible />).props();
      expect(props).toEqual(defaultProps);
    });
  });
});
