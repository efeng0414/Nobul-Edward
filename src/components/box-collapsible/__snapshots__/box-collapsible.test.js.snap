// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxCollapsible ComponentSnapshot should match existing snapshot 1`] = `
<div
  className="box-collapsible-holder component"
>
  <input
    className="box-collapsible-toggle"
    defaultChecked={true}
    id="collapsible"
    type="checkbox"
  />
  <div
    className="box-collapsible"
  >
    <label
      className="box-collapsible-label"
      htmlFor="collapsible"
    />
    <div
      className="box-collapsible-content"
    >
      <div
        className="box-collapsible-content-inner"
      >
        abc
      </div>
    </div>
  </div>
</div>
`;
